### PR TITLE
rux-table - add spacing tokens and adjust to fit astro 7.0 design specs

### PR DESCRIPTION
**Pull request by kiley-mitti**
## Brief Description

Adjusted rux-table and all sub components to fit astro 7.0 design specifications. Also added spacing tokens.

## JIRA Link

[ASTRO-4606](https://rocketcom.atlassian.net/browse/ASTRO-4606)

## Motivation and Context

Part of the Astro 7.0 Spacing initiative!

## Types of changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

## Checklist

- [ ] This PR adds or removes a Storybook story.
- [ ] I have added tests to cover my changes.
- [x] Regressions are passing and/or failures are documented
- [x] Changes have been checked in evergreen browsers

### CODE REVIEW

File: packages/web-components/src/components/rux-table/rux-table-cell/rux-table-cell.scss [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
     border-style: solid;
     border-color: var(--color-background-base-default);
-    padding: 0.625rem 1rem;
+    font-family: var(--font-body-1-font-family);
+    font-size: var(--font-body-1-font-size);
+    font-weight: var(--font-body-1-font-weight);
+    letter-spacing: var(--font-body-1-letter-spacing);
+    line-height: var(--font-body-1-line-height);
+    padding: var(--spacing-0) var(--spacing-2);
+    height: calc(var(--spacing-8) - 1px); //32 (height and borders)
```
Comment (FMorrison87) [inline packages/web-components/src/components/rux-table/rux-table-cell/rux-table-cell.scss:13]: I had some deep discussions with design while you were gone and I learned that when there is documentation in the design that indicates a fixed height, this is a design directive, not a dev one. I took a look at the designs in Figma and it does specify this height, but it says minimum, so perhaps change this to min-height? Hopefully that doesn't mess up the styling.

File: packages/web-components/src/components/rux-table/rux-table-header-cell/rux-table-header-cell.scss [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
     background: var(--color-background-surface-header);
     color: var(--color-text-primary);
     font-family: var(--font-heading-5-font-family);
     font-size: var(--font-heading-5-font-size);
     font-weight: var(--font-heading-5-font-weight);
     letter-spacing: var(--font-heading-5-letter-spacing);
-    padding: 0.625rem 1rem;
+    line-height: var(--font-heading-5-line-height);
+    padding: var(--spacing-0) var(--spacing-2);
+    height: var(--spacing-10); //40
```
Comment (FMorrison87) [inline packages/web-components/src/components/rux-table/rux-table-header-cell/rux-table-header-cell.scss:12]: Same thing for this guy, min-height of 40px.

**Review by FMorrison87 — DISMISSED** [review]

File: packages/web-components/src/components/rux-table/rux-table-cell/rux-table-cell.scss (hunk 1 above)
Comment (kiley-mitti) [reply packages/web-components/src/components/rux-table/rux-table-cell/rux-table-cell.scss:13]: I had it at min height at first but display:table doesn't respect min-height like other displays do. It treats height as min-height as far as I know. :)

File: packages/web-components/src/components/rux-table/rux-table-header-cell/rux-table-header-cell.scss (hunk 2 above)
Comment (kiley-mitti) [reply packages/web-components/src/components/rux-table/rux-table-header-cell/rux-table-header-cell.scss:12]: Same as above :)

**Review by FMorrison87 — APPROVED** [review]

**Review by markacianfrani — APPROVED** [review]

**Pull request merged**